ta4j-core/src/main/java/org/ta4j/core/num/PrecisionNum.java

```diff
@@ -290,7 +292,14 @@ public Num sqrt(int precision) {
             BigDecimal estimatedExponent = exponent.divide(new BigDecimal(2));
             String estimateString = String.format("%sE%s", estimatedMantissa, estimatedExponent);
             log.trace("x[0] =~ sqrt({}...*10^{}) =~ {}", mantissa, exponent, estimateString);
-            estimate = new BigDecimal(estimateString);
+            DecimalFormat format = new DecimalFormat();
+            format.setParseBigDecimal(true);
+            try {
+                estimate = (BigDecimal) format.parse(estimateString);
+            } catch (ParseException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
```

Comment:
It is not good idea to print exception to the console during precision calculation.
will be better to wrap it into kind of RuntimeException.
in this case we can come when estimate stay still as a previous value

Reply:
`DecimalFormat.parse(String)` throws `Exception`.  Wrap with `try/catch` or `Num sqrt() throws Exception`?

Reply:
Yes, no imaginable way to handle this should throw RuntimeException("Could not format Number", e),

Reply:
1.
throw RuntimeException("Unable to parse into the DecimalFormat the following string: " + estimateString, e);
actually we don't care for original ParseException, so it can be even without ",e"
2. static code analyzer gives bad mark for this approach :)

Reply:
I would preffer using a more explicit message like @laviua suggests because I cannot exculding any potential fault. If there will be any errors the programm should stop and not calculate wrong results whoes cause is not transparent.